lib/tlog/play.c

```diff
@@ -924,9 +924,17 @@ tlog_play_run(struct tlog_errs **perrs, int *psignal)
             /* If hit the end of stream */
             if (tlog_pkt_is_void(&pkt)) {
                 tlog_play_goto_active = false;
-                if (tlog_play_follow) {
-                    read_wait = (struct timespec){POLL_PERIOD, 0};
-                    continue;
+                if (tlog_play_paused) {
+                    do {
+                        rc = clock_nanosleep(CLOCK_MONOTONIC, 0,
+                                             &tlog_timespec_max, NULL);
```

Comment:
We shouldn't be copying pausing logic here. See if you can avoid that. It would be best if pausing would still be handled by the same (maybe expanded) code.